projects/noloan2/app/src/main/java/hasadna/noloan2/MainActivity.java

```diff
@@ -5,91 +5,58 @@
 import android.database.Cursor;
 import android.net.Uri;
 import android.os.Bundle;
-import android.support.annotation.NonNull;
 import android.support.v7.app.AppCompatActivity;
-import android.support.v7.widget.LinearLayoutManager;
-import android.support.v7.widget.RecyclerView;
-import android.util.Base64;
-import android.view.View;
-import android.widget.Button;
-import android.widget.EditText;
+import android.util.Log;
+import android.widget.TextView;
 import android.widget.Toast;
 
-import com.example.galgo.noloan.protobuf.UserProto.loguser;
-import com.google.firebase.firestore.CollectionReference;
-import com.google.firebase.firestore.FirebaseFirestore;
-import com.google.firebase.firestore.Query;
-
 import java.util.ArrayList;
 
-import hasadna.noloan2.protobuf.SmsProto.SMSmessage;
+import hasadna.noloan2.protobuf.SMSProto.SmsMessage;
 import noloan.R;
 
-public class MainActivity extends AppCompatActivity implements View.OnClickListener {
-  
-  private static final int PERMISSION_REQUEST_CODE = 123;
+public class MainActivity extends AppCompatActivity {
   
-  EditText name;
-  RecyclerView userView;
+  TextView text;
   
-  FirebaseFirestore firestoreClient;
-  Query query;
+  ArrayList<SmsMessage> spamList;
   
   @Override
   protected void onCreate(Bundle savedInstanceState) {
     super.onCreate(savedInstanceState);
     setContentView(R.layout.activity_main);
     
-    initFirestore();
+    text = findViewById(R.id.TV_text);
     
-    name = findViewById(R.id.ET_name);
+    spamList = new ArrayList<>();
+    ArrayList<SmsMessage> messages = readSms();
     
-    Button add = findViewById(R.id.B_adduser);
-    add.setOnClickListener(this);
+    // For testing, adding the first message rad to the list of spam.
+    spamList.add(SmsMessage.newBuilder().setSender(messages.get(0).getSender()).setBody(messages.get(0).getBody()).build());
     
-    userView = findViewById(R.id.RV_userview);
-    userView.setAdapter(new SmsAdapter(readSms()));
-    // Firestore test
-    /*UserAdapter adapter = new UserAdapter(query);
-    userView.setAdapter(adapter);
-    adapter.startListening();*/
-    userView.setLayoutManager(new LinearLayoutManager(this));
+    int count = 0;
+    for (SmsMessage message : messages) {
+      for (SmsMessage spam : spamList) {
+        if (comparingMessages(message, spam)) {
+          count++;
+          break;
+        }
+      }
+    }
     
+    text.setText(count + "");
   }
   
-  private void initFirestore() {
-    firestoreClient = FirebaseFirestore.getInstance();
-    query = firestoreClient.collection("Users");
-  }
-  
-  // Write the user to the Firestore
-  private void writeUser(loguser user) {
-    FirestoreElement userElement = encodeUser(user);
+  // Basic comparing of two SMS massages
```

Comment:
Is the `equals()` of `SmsMessage` not adequate?

Reply:
Didn't know Proto has equals. will use it instand.